In [555]:
library(class)

<html><b>Reading data.</b></html>

In [556]:
data <- read.csv(file= "June 10-July 12, 2015 - Gaming, Jobs and Broadband - CSV.csv")

<html><b>Selecting five columns.</b></html>

In [557]:
data <- data[,c("age","sex","emplnw","par","inc")]

#"date1a", "emptype1","emptype3",

In [558]:
data <- na.omit(data)

<html><b>Categorising age.</b></html>

In [559]:
data$ageCat <- cut(data$age,c(18,25,50,80))
head(data)

age sex emplnw par inc ageCat 
1 47  2   4      1   99  (25,50]
2 63  2   3      2    6  (50,80]
3 86  2   3      2    3  NA     
4 40  2   1      2    6  (25,50]
5 65  2   3      2    3  (50,80]
6 69  2   2      2    8  (50,80]

In [560]:
na.omit(data)

age sex emplnw par inc ageCat 
1    47  2   4      1   99  (25,50]
2    63  2   3      2    6  (50,80]
4    40  2   1      2    6  (25,50]
5    65  2   3      2    3  (50,80]
6    69  2   2      2    8  (50,80]
7    72  1   3      2   99  (50,80]
8    70  1   3      2    7  (50,80]
9    77  1   3      2    6  (50,80]
10   73  1   3      2    3  (50,80]
12   77  2   3      1   99  (50,80]
13   52  1   1      2    8  (50,80]
14   74  2   3      2    6  (50,80]
15   43  2   1      1   98  (25,50]
16   44  2   1      2    7  (25,50]
17   68  1   3      2    4  (50,80]
19   68  2   3      2   99  (50,80]
20   58  1   1      2    7  (50,80]
21   31  2   1      1    2  (25,50]
24   57  1   1      2    6  (50,80]
25   70  1   3      2    9  (50,80]
26   36  2   4      1    7  (25,50]
27   39  2   4      1    6  (25,50]
29   76  2   3      2    6  (50,80]
30   63  2   1      2    6  (50,80]
31   66  2   3      2    5  (50,80]
32   57  2   3      2    7  (50,80]
33   19  1   2      2    2  (18,25]
34   28  1   2      2    3  (25,50]
35   49  2   4      1    8  (25,50]
36   45  2   1      1    4  (25,50]
...  ... ... ...    ... ... ...    
1972 27  2   4      1    1  (25,50]
1973 66  2   3      2    5  (50,80]
1974 23  1   1      2    4  (18,25]
1975 79  1   3      2    4  (50,80]
1976 29  2   4      1    5  (25,50]
1977 65  2   2      1   99  (50,80]
1978 35  2   1      2    5  (25,50]
1979 26  1   2      2    1  (25,50]
1980 60  1   1      2    9  (50,80]
1981 51  2   6      2    1  (50,80]
1982 27  1   4      1    1  (25,50]
1983 60  2   2      2   99  (50,80]
1984 43  1   4      2    3  (25,50]
1985 19  2   2      2    6  (18,25]
1986 36  2   2      1    1  (25,50]
1987 62  1   1      2    8  (50,80]
1988 59  1   1      2    8  (50,80]
1989 34  2   4      1    6  (25,50]
1990 56  1   1      2    5  (50,80]
1991 63  1   1      2    3  (50,80]
1992 50  1   1      2    3  (25,50]
1993 47  2   1      2    5  (25,50]
1994 72  1   3      2    5  (50,80]
1995 70  2   3      2    4  (50,80]
1996 35  1   1      2    1  (25,50]
1997 39  2   4      1    9  (25,50]
1998 77  2   3      2    3  (50,80]
1999 40  2   2      1    6  (25,50]
2000 54  2   1      2    4  (50,80]
2001 31  2   4      1    5  (25,50]

<html><b>converting data to numeric.</b></html>

In [561]:
data$ageCat <- as.numeric(data$ageCat)

<html><b>Labeling the age category.</b></html>

In [562]:
if((data$ageCat > 18 ) && (data$ageCat <=25)){
    data$ageCat <- 1
}
if((data$ageCat > 25 ) && (data$ageCat <=50)){
    data$ageCat <- 2
}
if((data$ageCat > 50 ) && (data$ageCat <=80)){
    data$ageCat <- 3
}

In [563]:
data$sex <- as.numeric(data$sex)

In [564]:
data <-data[,c("age","sex","emplnw","par","inc")] 
data <- na.omit(data)

<html><b>Normalizing Data.</b></html>

In [565]:
data$age <- (data$age - min(data$age)) /(max(data$age)-min(data$age))
data$sex <- (data$sex - min(data$sex)) /(max(data$sex)-min(data$sex))
data$emplnw <- (data$emplnw - min(data$emplnw)) /(max(data$emplnw)-min(data$emplnw))
data$par <- (data$par - min(data$par)) /(max(data$par)-min(data$par))
data$inc <- (data$inc - min(data$inc)) /(max(data$inc)-min(data$inc))

In [566]:
data

age        sex emplnw     par   inc       
1    0.3580247  1   0.03061224 0.000 1.00000000
2    0.5555556  1   0.02040816 0.125 0.05102041
3    0.8395062  1   0.02040816 0.125 0.02040816
4    0.2716049  1   0.00000000 0.125 0.05102041
5    0.5802469  1   0.02040816 0.125 0.02040816
6    0.6296296  1   0.01020408 0.125 0.07142857
7    0.6666667  0   0.02040816 0.125 1.00000000
8    0.6419753  0   0.02040816 0.125 0.06122449
9    0.7283951  0   0.02040816 0.125 0.05102041
10   0.6790123  0   0.02040816 0.125 0.02040816
11   0.7777778  1   0.02040816 0.125 1.00000000
12   0.7283951  1   0.02040816 0.000 1.00000000
13   0.4197531  0   0.00000000 0.125 0.07142857
14   0.6913580  1   0.02040816 0.125 0.05102041
15   0.3086420  1   0.00000000 0.000 0.98979592
16   0.3209877  1   0.00000000 0.125 0.06122449
17   0.6172840  0   0.02040816 0.125 0.03061224
18   0.8518519  0   0.02040816 0.000 0.03061224
19   0.6172840  1   0.02040816 0.125 1.00000000
20   0.4938272  0   0.00000000 0.125 0.06122449
21   0.1604938  1   0.00000000 0.000 0.01020408
22   0.8271605  0   0.02040816 0.125 0.05102041
23   0.9259259  1   0.02040816 0.125 0.00000000
24   0.4814815  0   0.00000000 0.125 0.05102041
25   0.6419753  0   0.02040816 0.125 0.08163265
26   0.2222222  1   0.03061224 0.000 0.06122449
27   0.2592593  1   0.03061224 0.000 0.05102041
28   0.8271605  1   0.02040816 0.125 0.00000000
29   0.7160494  1   0.02040816 0.125 0.05102041
30   0.5555556  1   0.00000000 0.125 0.05102041
...  ...        ... ...        ...   ...       
1972 0.11111111 1   0.03061224 0.000 0.00000000
1973 0.59259259 1   0.02040816 0.125 0.04081633
1974 0.06172840 0   0.00000000 0.125 0.03061224
1975 0.75308642 0   0.02040816 0.125 0.03061224
1976 0.13580247 1   0.03061224 0.000 0.04081633
1977 0.58024691 1   0.01020408 0.000 1.00000000
1978 0.20987654 1   0.00000000 0.125 0.04081633
1979 0.09876543 0   0.01020408 0.125 0.00000000
1980 0.51851852 0   0.00000000 0.125 0.08163265
1981 0.40740741 1   0.05102041 0.125 0.00000000
1982 0.11111111 0   0.03061224 0.000 0.00000000
1983 0.51851852 1   0.01020408 0.125 1.00000000
1984 0.30864198 0   0.03061224 0.125 0.02040816
1985 0.01234568 1   0.01020408 0.125 0.05102041
1986 0.22222222 1   0.01020408 0.000 0.00000000
1987 0.54320988 0   0.00000000 0.125 0.07142857
1988 0.50617284 0   0.00000000 0.125 0.07142857
1989 0.19753086 1   0.03061224 0.000 0.05102041
1990 0.46913580 0   0.00000000 0.125 0.04081633
1991 0.55555556 0   0.00000000 0.125 0.02040816
1992 0.39506173 0   0.00000000 0.125 0.02040816
1993 0.35802469 1   0.00000000 0.125 0.04081633
1994 0.66666667 0   0.02040816 0.125 0.04081633
1995 0.64197531 1   0.02040816 0.125 0.03061224
1996 0.20987654 0   0.00000000 0.125 0.00000000
1997 0.25925926 1   0.03061224 0.000 0.08163265
1998 0.72839506 1   0.02040816 0.125 0.02040816
1999 0.27160494 1   0.01020408 0.000 0.05102041
2000 0.44444444 1   0.00000000 0.125 0.03061224
2001 0.16049383 1   0.03061224 0.000 0.04081633

In [567]:
set.seed(76964057)

<html><b>Performing k-means for number of clusters 3.</b></html>

In [568]:
k <-kmeans(data, centers=3, algorithm=c("Hartigan-Wong")) #Create 3 clusters

In [569]:
nrow(data)

[1] 2001

In [570]:
k

K-means clustering with 3 clusters of sizes 989, 173, 839

Cluster means:
        age sex     emplnw        par        inc
1 0.3984821   0 0.01584780 0.10288170 0.17170508
2 0.5654749   1 0.02842987 0.13945087 0.99593016
3 0.4179579   1 0.01849870 0.09579857 0.03952713

Clustering vector:
   1    2    3    4    5    6    7    8    9   10   11   12   13   14   15   16 
   2    3    3    3    3    3    1    1    1    1    2    2    1    3    2    3 
  17   18   19   20   21   22   23   24   25   26   27   28   29   30   31   32 
   1    1    2    1    3    1    3    1    1    3    3    3    3    3    3    3 
  33   34   35   36   37   38   39   40   41   42   43   44   45   46   47   48 
   1    1    3    3    3    1    3    1    3    3    3    3    1    1    1    2 
  49   50   51   52   53   54   55   56   57   58   59   60   61   62   63   64 
   1    3    3    1    3    2    3    3    1    3    3    3    3    2    1    1 
  65   66   67   68   69   70   71   72   73   74   75   76   

In [571]:
k$centers

age       sex emplnw     par        inc       
1 0.3984821 0   0.01584780 0.10288170 0.17170508
2 0.5654749 1   0.02842987 0.13945087 0.99593016
3 0.4179579 1   0.01849870 0.09579857 0.03952713

In [572]:
table(k$cluster)


  1   2   3 
989 173 839 

In [573]:
rng<-2:20 #K from 2 to 20
tries<-100 #Run the K Means algorithm 100 times
avg.totw.ss<-integer(length(rng)) #Set up an empty vector to hold all of points

<html><b>Taking the two clustering values.</b></html>

In [574]:
v <- c(3,5)

<html><b>Trying other values for K</b></html>

In [575]:
for(j in 1:2){ # For each value of the range variable
v.totw.ss<-integer(tries) #Set up an empty vector to hold the 100 tries
for(i in 1:tries){
k.temp<-kmeans(data,centers=v[j]) #Run kmeans
v.totw.ss[i]<-k.temp$tot.withinss#Store the total withinss
}
avg.totw.ss[v-1]<-mean(v.totw.ss) #Average the 100 total withinss
}


<html><b>Performing k-means for number of clusters 5.</b></html>

In [576]:
k <-kmeans(data, centers=5,algorithm=c("Hartigan-Wong"))
table(k$cluster)
k$centers


  1   2   3   4   5 
173 839 466 391 132 

age       sex emplnw     par        inc       
1 0.5654749 1   0.02842987 0.13945087 0.99593016
2 0.4179579 1   0.01849870 0.09579857 0.03952713
3 0.5595295 0   0.01344486 0.11292918 0.04618113
4 0.1750813 0   0.00871653 0.07544757 0.04293022
5 0.4916760 0   0.04545455 0.14867424 0.99628942

<html><b>Interpretation of the cluster characteristics.
<p> Cluster: 3<br>
Cluster 1 contains 989 points. Most of the people of the age belong to 76 years category.  <br>Cluster 2 contains 173 points. Most of the people of that category belongs to around 54 years. <br> Cluster 3 contains 839 points. Most of the people belong to around 30 years of age.</p>

<p> Cluster: 5<br>
Cluster 1 contains 173 points. Cluster 2 contains 839 points. Cluster 3 contains 466 points. Cluster 4 contains 391 points. Cluster 5 contains 132 points.</p>
</b></html>